topo/node/cptx/cptx.go

```diff
@@ -26,18 +39,134 @@ func New(pb *topopb.Node) (node.Implementation, error) {
 
 type Node struct {
 	pb *topopb.Node
+	cliConn *scraplinetwork.Driver
 }
 
 func (n *Node) Proto() *topopb.Node {
 	return n.pb
 }
 
+// WaitCLIReady attempts to open the transport channel towards a Network OS and perform scrapligo OnOpen actions
+// for a given platform. Retries with exponential backoff.
+func (n *Node) WaitCLIReady(sleep time.Duration, attempts int) error {
+    var err error = nil
+    for i := 0; i < attempts; i++ {
+        if i > 0 {
+			log.Debugf("%s - cli not ready - waiting.", n.pb.Name)
+            time.Sleep(sleep)
+            sleep *= 2
+        }
+		err = n.cliConn.Open()
+        if err == nil {
+		    log.Debugf("%s - cli ready.", n.pb.Name)
+	        return nil
+		}
+	}
+    return err
+}
+
+// PatchCLIConnOpen sets the OpenCmd and ExecCmd of system transport to work with `kubectl exec` terminal.
+func (n *Node) PatchCLIConnOpen(ns string) error {
+	t, ok := n.cliConn.Transport.Impl.(scraplitransport.SystemTransport)
+	if !ok {
+		return ErrIncompatibleCliConn
+	}
+
+	t.SetExecCmd("kubectl")
+	t.SetOpenCmd([]string{"exec", "-it", "-n", ns, n.pb.Name, "--", "cli", "-c"})
+
+	return nil
+}
+
+// SpawnCLIConn spawns a CLI connection towards a Network OS using `kubectl exec` terminal and ensures CLI is ready
+// to accept inputs.
+func (n *Node) SpawnCLIConn(ns string) error {
+	d, err := scraplicore.NewCoreDriver(
+		n.pb.Name,
+		"juniper_junos",
+		scraplibase.WithAuthBypass(true),
+		// disable transport timeout
+		scraplibase.WithTimeoutTransport(0),
+	)
+	if err != nil {
+		return err
+	}
+
+	n.cliConn = d
+
+	err = n.PatchCLIConnOpen(ns)
+	if err != nil {
+		n.cliConn = nil
+
+		return err
+	}
+
+	err = n.WaitCLIReady(1, max_retries)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 func (n *Node) GenerateSelfSigned(ctx context.Context, ni node.Interface) error {
         return status.Errorf(codes.Unimplemented, "Unimplemented")
 }
 
 func (n *Node) ConfigPush(ctx context.Context, ns string, r io.Reader) error {
```

Comment:
can you please add unittest coverage for this function

Reply:
Done. Do check my other comment.